presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/api/ControllerConfigAPI.java

```diff
@@ -0,0 +1,181 @@
+    }
+
+    @POST
+    @Path("/{file}")
+    @Produces({MediaType.TEXT_PLAIN})
+    public Response setConfigFileByURL(
+            @PathParam("file") @ApiParam("The name of a file") String file,
+            String url,
+            @QueryParam("scope") String scope,
+            @QueryParam("nodeId") List<Integer> nodeId)
+    {
+        ApiRequester apiRequester = ApiRequester.builder(ControllerConfigAPI.class)
+                .pathMethod("setConfigFileByURL")
+                .httpMethod(POST)
+                .resolveTemplate("file", file)
+                .accept(MediaType.TEXT_PLAIN)
+                .entity(Entity.entity(url, MediaType.TEXT_PLAIN))
+                .build();
+
+        return getResponseFromRequest(scope, apiRequester, nodeId);
+    }
+
+    @DELETE
+    @Path("/{file}")
+    @Produces({MediaType.TEXT_PLAIN})
+    public Response deleteConfigFile(
+            @PathParam("file") @ApiParam("The name of a file") String file,
+            @QueryParam("scope") String scope,
+            @QueryParam("nodeId") List<Integer> nodeId)
+    {
+        ApiRequester apiRequester = ApiRequester.builder(ControllerConfigAPI.class)
+                .pathMethod("deleteConfigFile")
+                .httpMethod(DELETE)
+                .resolveTemplate("file", file)
+                .accept(MediaType.TEXT_PLAIN)
+                .build();
+
+        return getResponseFromRequest(scope, apiRequester, nodeId);
+    }
+
+    private Response getResponseFromRequest(String scope, ApiRequester apiRequester, Collection<Integer> nodeId)
+    {
+        if (((scope != null) && (!nodeId.isEmpty())) || (scope == null && nodeId.isEmpty())) {
+            LOGGER.error("Invalid parameters");
+            return Response.status(BAD_REQUEST).entity("Invalid parameters").build();
+        }
+
+        ApiScope apiScope;
+        try {
+            apiScope = ApiScope.fromString(scope);
+        }
+        catch (IllegalArgumentException e) {
+            return Response.status(BAD_REQUEST).entity("Invalid scope").build();
+        }
+
+        Collection<URI> uriCollection;
+        try {
+            uriCollection = nodeId.isEmpty() ? nodeSet.getUris(apiScope) : nodeSet.getUrisByIds(nodeId);
+        }
+        catch (IllegalArgumentException e) {
+            LOGGER.error("Invalid node ID");
+            return Response.status(BAD_REQUEST).entity("Invalid node ID").build();
+        }
+
+        if (apiScope == ApiScope.COORDINATOR && uriCollection.size() != 1) {
+            LOGGER.error("Number of coordinator is not 1");
+            return Response.status(INTERNAL_SERVER_ERROR).entity("Number of coordinator is not 1").build();
+        }
+
+        List<Response> responseList = new ArrayList<>();
+        for (URI uri : uriCollection) {
+            responseList.add(apiRequester.send(uri));
+        }
+        return Response.status(MULTI_STATUS)
+                .entity(new Gson().toJson(responseList)).type(MediaType.APPLICATION_JSON).build();
```

Comment:
We should either have agents include their IDs in their responses, or make this a `Map<NodeID, Response>`.